compiler/IREmitter/Payload/patches/vm_insnhelper.c

```diff
@@ -481,3 +484,232 @@ void sorbet_throwReturn(rb_execution_context_t *ec, VALUE retval) {
     ec->errinfo = v;
     EC_JUMP_TAG(ec, ec->tag->state);
 }
+
+#define SORBET_INLINE __attribute__((always_inline))
+
+SORBET_INLINE
+static int computeLocalIndex(long index) {
+    // Local offset calculation needs to take into account the fixed values that
+    // are present on the stack:
+    // https://github.com/ruby/ruby/blob/a9a48e6a741f048766a2a287592098c4f6c7b7c7/compile.c#L1509
+    return index + VM_ENV_DATA_SIZE;
+}
+
+SORBET_INLINE
+static VALUE sorbet_readLocal(rb_control_frame_t *cfp, long index, long level) {
+    int offset = computeLocalIndex(index);
+    return *(vm_get_ep(cfp->ep, level) - offset);
+}
+SORBET_INLINE
+static void sorbet_writeLocal(rb_control_frame_t *cfp, long index, long level, VALUE value) {
+    int offset = computeLocalIndex(index);
+    vm_env_write(vm_get_ep(cfp->ep, level), -offset, value);
+}
```

Comment:
Thanks for marking all these functions `static`, it would be really bad if generated code started using them instead of the versions in `codegen-payload.c`. Maybe it would be worth adding a comment here to indicate that these are intentionally duplicated? Maybe different names would be better?